pkg/nodejs/npm/parse.go

```diff
@@ -117,8 +119,13 @@ func (p *Parser) parseV2(packages map[string]Package) ([]types.Library, []types.
 		}
 		libs[pkgID] = lib
 
-		dependsOn := make([]string, 0, len(pkg.Dependencies))
-		for depName, depVersion := range pkg.Dependencies {
+		// npm builds graph using optional deps. e.g.:
+		// └─┬ watchpack@1.7.5
+		// ├─┬ chokidar@3.5.3 - optional dependency
+		// │ └── glob-parent@5.1.
+		dependencies := lo.Assign(pkg.Dependencies, pkg.OptionalDependencies)
```

Comment:
Did you add a test for optional dependencies?

Reply:
We can update the existing test case.

Reply:
Added optional deps to testcases